go/logictest/runner.go

```diff
@@ -238,19 +272,67 @@ func runTestFile(harness Harness, file string) {
 		panic(err)
 	}
 
+	dnr := false
+	ll := newLoggingLock()
+
 	for _, record := range testRecords {
-		_, _, cont, _ := executeRecord(harness, record)
-		if !cont {
-			break
+		currRecord = record
+		startTime = time.Now()
+
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		lockCtx := context.WithValue(ctx, "lock", ll)
+
+		if dnr {
```

Comment:
Would return after this log statement to reduce nesting

Reply:
now using `continue`